network/src/protocols/discovery/mod.rs

```diff
@@ -378,6 +373,22 @@ where
     }
 }
 
+/// The note which has been verified its validity
+struct VerifiedNote {
+    peer_id: PeerId,
+    peer_info: VerifiedPeerInfo,
+    /// the raw `Note` sent from remote
+    raw: Note,
+}
+
+/// The peer info which has been verified its validity
+struct VerifiedPeerInfo {
+    addrs: Vec<Multiaddr>,
+    epoch: u64,
+    /// the raw `PeerInfo` sent from remote
+    raw: PeerInfo,
```

Comment:
I don't think there's a need for storing the unverified raw peer info. The raw note is needed because that is what we forward to other peers (in `fn reconcile`). But the raw peer info is never used as such.

Reply:
In fact, I now feel like we don't need 2 separate structs at all. Just one flattened struct is enough IMO.
`known_peers` can be of type `HashMap<PeerId, VerifiedNote>`

```
struct VerifiedNote {
   peer_id: PeerId,
   addrs: Vec<Multiaddr>,
   epoch: u64,
   raw_note: Note,
}
```

Reply:
That makes sense. Thank you!